src/main/java/bio/terra/common/GetResourceBufferProjectStep.java

```diff
@@ -29,12 +34,31 @@ public GetResourceBufferProjectStep(
   @Override
   public StepResult doStep(FlightContext context) {
     FlightMap workingMap = context.getWorkingMap();
-    ResourceInfo resource = bufferService.handoutResource(enableSecureMonitoring);
-    String projectId = resource.getCloudResourceUid().getGoogleProjectUid().getProjectId();
-    logger.info("Retrieved project from RBS with ID: {}", projectId);
+    try {
+      ResourceInfo resource = bufferService.handoutResource(enableSecureMonitoring);
+      String projectId = resource.getCloudResourceUid().getGoogleProjectUid().getProjectId();
+      logger.info("Retrieved project from RBS with ID: {}", projectId);
 
-    workingMap.put(ProjectCreatingFlightKeys.GOOGLE_PROJECT_ID, projectId);
-    return StepResult.getStepResultSuccess();
+      workingMap.put(ProjectCreatingFlightKeys.GOOGLE_PROJECT_ID, projectId);
+      return StepResult.getStepResultSuccess();
+    } catch (BufferServiceAPIException e) {
```

Comment:
This code doesn't exactly match what WSM is doing, from looking at https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/service/workspace/flight/cloud/gcp/PullProjectFromPoolStep.java#L48

This code is handling `INTERNAL_SERVER_ERROR` as a retryable failure but the WSM code isn't, is that intentional?

Reply:
(We've been chatting about this offline, but just to write it down) -- Yeah, I've added the retry on the internal_server_error because we're still seeing the server errors coming from RBS. I'm not sure if this is the best approach, but wanted to test if the retry here helped with test flakiness.